src/mdlComponentHandler.js

```diff
@@ -390,7 +366,16 @@ componentHandler = (function() {
      * @param  {!Node} node the node to be downgraded
      */
     var downgradeNode = function(node) {
-      deconstructComponentInternal(findCreatedComponentByNodeInternal(node));
+      var found = [];
+      for (var n = 0; n < createdComponents_.length; n++) {
```

Comment:
We talked about using it before in the componentHandler. It was decided that we were already using for loops so we should continue with that to be consistent in the codebase instead of converting over to forEach. I also think the scenario of someone hacking it to work in IE9 also came up. This was done before we even launched iirc, or very shortly there-after.